tests/functional/settings_secondary_emails.js

```diff
@@ -60,6 +60,59 @@ define([
       return this.remote.then(clearBrowserState());
     },
 
+    'gated in unverified session open verification same tab': function () {
+      return this.remote
+        // when an account is created, the original session is verified
+        // re-login to destroy original session and created an unverified one
+        .then(openPage(SIGNUP_URL, selectors.SIGNUP.HEADER))
+        .then(fillOutSignUp(email, PASSWORD))
+        .then(testElementExists(selectors.CONFIRM_SIGNUP.HEADER))
+        .then(openVerificationLinkInSameTab(email, 0))
+        .then(testElementExists(selectors.SETTINGS.HEADER))
+        .then(click(selectors.SETTINGS.SIGNOUT))
+        .then(testElementExists(selectors.SIGNIN.HEADER))
+        .then(fillOutSignIn(email, PASSWORD))
+        .then(testElementExists(selectors.SETTINGS.HEADER))
+
+        // unlock panel
+        .then(click(selectors.UNLOCK.UNLOCK_BUTTON))
```

Comment:
`UNLOCK.UNLOCK_BUTTON` should be made more specific, it'll break as soon as we have multiple panels that can be unlocked. Can you use the already existing `EMAIL` namespace?